File: examples/primitives/swift/src/inheritance/square.swift
import Foundation

import CWrapper

/**
 */
public class Square : IRhombus, IRectangle {
    
    public let cself : UnsafeMutableRawPointer
    public let owner : Bool

    // internal main initializer
    internal required init(_ _cself: UnsafeMutableRawPointer , _ _owner: Bool = false) {
      self.cself = _cself
      self.owner = _owner
    }
    deinit {
      if owner {
        release_Square(cself)
      }
    }
    
    /**
     */
    public convenience init(side: Double) {

        
        self.init(create_Square(side), true)
    }
    
    /**
     */
    public var area: Double {
          get {
            let result = _prop_get_Square_area(cself);
            
            return result
          }
          
    }
    /**
     */
    public func perimeter() -> Double {

        let result = _func_Square_perimeter(cself);
        
        return result;
    }
}

File: examples/primitives/kotlin/src/main/java/com/examples/inheritance/square.kt
package com.examples.inheritance

import alias.*


class Square
internal constructor(_id: Long) : IRhombus, IRectangle, AutoCloseable {
    companion object {
        init {
            System.loadLibrary("wrapper_jni");
        }
        
        protected fun construct_helper(side: Double): Long {
            val id = jConstructor(side)
            return id
        }

        @JvmStatic
        private external fun jConstructor(side: Double): Long

    }
    
    protected var id = _id
    override fun getObjId(): Long {
        if (id == 0L) {
            throw RuntimeException("Object is not allocated")
        }
        return id;
    }
    
    constructor(side: Double): this(construct_helper(side)) {
        //jSet_this(id, this)
    }
    
    override val area: Double
        get() {
            val result = jArea(getObjId())
            
            return result
        }
    
    override fun perimeter(): Double {
        val result = jPerimeter(getObjId())
        
        return result
    }

    override fun close() {
        if (id != 0L) {
            jFinalize(id)
            id = 0L
        }
    }

    /**
     * Finalize and deletes the object
     */
    protected fun finalize() {
        close()
    }

    ///// External wrapper functions ////////////
    private external fun jArea(id: Long): Double
    private external fun jPerimeter(id: Long): Double
    private external fun jSet_this(id: Long, self: Any): Unit
    private external fun jFinalize(id: Long): Unit
}